sdks/python/apache_beam/runners/interactive/dataproc/dataproc_cluster_manager.py

```diff
@@ -0,0 +1,154 @@
+          'No region information was detected, defaulting Dataproc cluster '
+          'region to: us-central1.')
+      self.region = 'us-central1'
+
+    if cluster_name:
+      _LOGGER.warning(
+          'A user-specified cluster_name has been detected. '
+          'Please note that you will have to manually delete the Dataproc '
+          'cluster that will be created under the name: %s',
+          cluster_name)
+      self.cluster_name = cluster_name
+    else:
+      self.cluster_name = self.DEFAULT_NAME
+
+    self.cluster_client = dataproc_v1.ClusterControllerClient(
+        client_options={
+            'api_endpoint': f'{self.region}-dataproc.googleapis.com:443'
+        })
+
+  def _create_cluster(self, cluster):
+    """Attempts to create a cluster using attributes that were
+    initialized with the DataprocClusterManager instance.
+
+    Args:
+      cluster: Dictionary representing Dataproc cluster
+    """
+    try:
+      self.cluster_client.create_cluster(
+          request={
+              'project_id': self.project_id,
+              'region': self.region,
+              'cluster': cluster
+          })
+      _LOGGER.info('Cluster created successfully: %s', self.cluster_name)
+    except Exception as e:
+      if e.code == 409:
+        if self.cluster_name == self.DEFAULT_NAME:
+          _LOGGER.info(
+              'Cluster %s already exists. Continuing...', self.DEFAULT_NAME)
+        else:
+          _LOGGER.error(
+              'Cluster already exists - unable to create cluster: %s',
+              self.cluster_name)
+          raise ValueError(
+              'Cluster {} already exists!'.format(self.cluster_name))
+      elif e.code == 403:
+        _LOGGER.error(
+            'Due to insufficient project permissions, '
+            'unable to create cluster: %s',
+            self.cluster_name)
+        raise ValueError(
+            'You cannot create a cluster in project: {}'.format(
+                self.project_id))
+      elif e.code == 501:
+        _LOGGER.error('Invalid region provided: %s', self.region)
+        raise ValueError('Region {} does not exist!'.format(self.region))
+      else:
+        _LOGGER.error('Unable to create cluster: %s', self.cluster_name)
+        raise e
+
+  # TODO(victorhc): Add support for user-specified pip packages
+  def create_flink_cluster(self):
+    """Calls _create_cluster with a configuration that enables FlinkRunner."""
+    cluster = {
+        'project_id': self.project_id,
+        'cluster_name': self.cluster_name,
+        'config': {
+            'software_config': {
+                'optional_components': ['DOCKER', 'FLINK']
+            }
+        }
+    }
+    self._create_cluster(cluster)
+
+  def cleanup(self):
+    """Deletes the cluster that uses the attributes initialized
+    with the DataprocClusterManager instance if the default
+    cluster_name is used."""
+    if self.cluster_name != self.DEFAULT_NAME:
+      return
+
+    try:
```

Comment:
This feels a bit unnatural. Since the `cluster_name` is already an attribute of the manager, this function is better rephrased into 2 functions:
- `delete_cluster(self)`: deletes the cluster
- `cleanup_if_default(self)`: checks the cluster_name and invoke `self.delete_cluster` if the name is default value.

Reply:
That would make more sense. Thanks for the suggestion!